x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/AbstractPhysicalOperationProviders.java

```diff
@@ -167,21 +167,25 @@ else if (aggregatorMode.isOutputPartial()) {
                 s -> aggregatorFactories.add(s.supplier.groupingAggregatorFactory(s.mode, s.channels)),
                 context
             );
+            int maxPageSize = context.pageSize(aggregateExec, aggregateExec.estimatedRowSize());
+            int aggregationBatchSize = maxPageSize; // TODO pick a more sensible number
```

Comment:
This is the number we were using previously. But it was never a very sane number. It's in the right ballpark, but it's probably better off being a constant or something.